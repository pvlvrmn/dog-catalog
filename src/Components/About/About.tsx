function About() {
  return (
    <div className="about__wr">
      <div className="about__title"><h1>About this project</h1></div>
      <div className="about__body">
        This project was built specifically for
        {' '}
        <a href="https://dog.ceo/dog-api/documentation/">Dog API</a>
        .
        <br />
        <br />
        <h2>Tech stack</h2>
        <ul className="about__stack">
          <li><a rel="noreferrer" href="https://react.dev/" target="_blank">React</a></li>
          <li><a rel="noreferrer" href="https://reactrouter.com/en/main" target="_blank">React Router</a></li>
          <li><a rel="noreferrer" href="https://www.typescriptlang.org/" target="_blank">Typescript</a></li>
          <li><a rel="noreferrer" href="https://vitejs.dev/" target="_blank">Vite</a></li>
          <li><a rel="noreferrer" href="https://eslint.org/" target="_blank">ESLint</a></li>
          <li><a rel="noreferrer" href="https://www.npmjs.com/" target="_blank">npm</a></li>
        </ul>
      </div>
    </div>
  );
}

export default About;
