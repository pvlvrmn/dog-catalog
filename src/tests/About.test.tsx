import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import About from '../Components/About/About';

describe('About page', () => {
    test('About Page text render', () => {
        render(<About />);
        const linkElem = screen.getByText('Tech stack');
        expect(linkElem).toBeInTheDocument();
    });

    test('Number of links', () => {
        render(<About />);
        const linkElem = screen.getByRole('a');
        expect(linkElem).toBeInTheDocument();
    })
})

