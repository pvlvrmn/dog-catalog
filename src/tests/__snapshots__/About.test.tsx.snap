// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page About renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="about__wr"
      >
        <div
          class="about__title"
        >
          <h1>
            About this project
          </h1>
        </div>
        <div
          class="about__body"
        >
          This project was built specifically for
           
          <a
            href="https://dog.ceo/dog-api/documentation/"
          >
            Dog API
          </a>
          .
          <br />
          <br />
          <h2>
            Tech stack
          </h2>
          <ul
            class="about__stack"
          >
            <li>
              <a
                href="https://react.dev/"
                rel="noreferrer"
                target="_blank"
              >
                React
              </a>
            </li>
            <li>
              <a
                href="https://reactrouter.com/en/main"
                rel="noreferrer"
                target="_blank"
              >
                React Router
              </a>
            </li>
            <li>
              <a
                href="https://www.typescriptlang.org/"
                rel="noreferrer"
                target="_blank"
              >
                Typescript
              </a>
            </li>
            <li>
              <a
                href="https://jestjs.io/"
                rel="noreferrer"
                target="_blank"
              >
                Jest
              </a>
            </li>
            <li>
              <a
                href="https://vitejs.dev/"
                rel="noreferrer"
                target="_blank"
              >
                Vite
              </a>
            </li>
            <li>
              <a
                href="https://eslint.org/"
                rel="noreferrer"
                target="_blank"
              >
                ESLint
              </a>
            </li>
            <li>
              <a
                href="https://www.npmjs.com/"
                rel="noreferrer"
                target="_blank"
              >
                npm
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="about__wr"
    >
      <div
        class="about__title"
      >
        <h1>
          About this project
        </h1>
      </div>
      <div
        class="about__body"
      >
        This project was built specifically for
         
        <a
          href="https://dog.ceo/dog-api/documentation/"
        >
          Dog API
        </a>
        .
        <br />
        <br />
        <h2>
          Tech stack
        </h2>
        <ul
          class="about__stack"
        >
          <li>
            <a
              href="https://react.dev/"
              rel="noreferrer"
              target="_blank"
            >
              React
            </a>
          </li>
          <li>
            <a
              href="https://reactrouter.com/en/main"
              rel="noreferrer"
              target="_blank"
            >
              React Router
            </a>
          </li>
          <li>
            <a
              href="https://www.typescriptlang.org/"
              rel="noreferrer"
              target="_blank"
            >
              Typescript
            </a>
          </li>
          <li>
            <a
              href="https://jestjs.io/"
              rel="noreferrer"
              target="_blank"
            >
              Jest
            </a>
          </li>
          <li>
            <a
              href="https://vitejs.dev/"
              rel="noreferrer"
              target="_blank"
            >
              Vite
            </a>
          </li>
          <li>
            <a
              href="https://eslint.org/"
              rel="noreferrer"
              target="_blank"
            >
              ESLint
            </a>
          </li>
          <li>
            <a
              href="https://www.npmjs.com/"
              rel="noreferrer"
              target="_blank"
            >
              npm
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
